### PR TITLE
new features and error parser

### PR DESCRIPTION
**Pull request by CustomIcon**
- added nsfw_classification
- updated to v2.2.5
- error parser updated.

### CODE REVIEW

File: coffeehouse/__init__.py [hunk 1]
```diff
@@ -1,16 +1,26 @@
...
-           "exception",
-           "lydia"] + api.__all__ + exception.__all__ + lydia.__all__
-__version__ = "2.1.0"
-__author__ = "Intellivoid Technologies"
-__source__ = "https://github.com/intellivoid/CoffeeHouse-Python-API-Wrapper"
-__copyright__ = "Copyright (c) 2017-2020 " + __author__
+__version__ = '2.2.5'
+__author__ = 'Intellivoid Technologies'
+__source__ = 'https://github.com/intellivoid/CoffeeHouse-Python-API-Wrapper'
+__copyright__ = 'Copyright (c) 2017-2021 ' + __author__
```
Comment (robbyoconnor) [inline coffeehouse/__init__.py:26]: Why not use an f-string here?

File: coffeehouse/api.py [hunk 2]
```diff
@@ -42,16 +45,24 @@ def _send(self, path, access_key=True, **payload):
...
         """
         if access_key:
-            payload["access_key"] = self.access_key
-        response = requests.post("{}/{}".format(self.endpoint, path), payload)
+            payload['access_key'] = self.access_key
+        response = requests.post(
+            '{}/{}'.format(
+                self.endpoint,
+                path,
+            ),
```
Comment (robbyoconnor) [inline coffeehouse/api.py:53]: Couldn't we use an f-string here?

File: examples/nsfw_example.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+# Reading the Image file and sending it through the API
+with open('path/to/image.jpg', 'rb') as f:
+    result = classifier.classify(f.read()).nsfw_classification
+
+# Output
+print('Content Hash: {}'.format(str(result.content_hash)))
+print('Content Type: {}'.format(str(result.content_type)))
+print('Safe Prediction: {}'.format(str(result.safe_prediction)))
+print('Unsafe Prediction: {}'.format(str(result.unsafe_prediction)))
+print('is NSFW: {}'.format(str(result.is_nsfw)))
```
Comment (robbyoconnor) [inline examples/nsfw_example.py:18]: Why not refactor to f-strings?

**Review by robbyoconnor — COMMENTED** [review]
Minor comments, more so questions but feel free to ignore if it's not feasible.

File: coffeehouse/__init__.py (hunk 1 above)
Comment (penn5) [reply coffeehouse/__init__.py:26]: Because it's easier not to

File: coffeehouse/api.py [hunk 4]
```diff
@@ -42,16 +45,24 @@ def _send(self, path, access_key=True, **payload):
...
+        )
         request_id = None
-        if "x-request-id" in response.headers:
-            request_id = response.headers["x-request-id"]
-        result = CoffeeHouseError.parse_and_raise(response.status_code,
-                                                  response.text,
-                                                  request_id)
-        if "payload" in result:
+        if 'x-request-id' in response.headers:
+            request_id = response.headers['x-request-id']
```
Comment (penn5) [inline coffeehouse/api.py:58]: request_id = response.headers.get("x-request-id", None)

File: coffeehouse/exception.py [hunk 5]
```diff
@@ -1,78 +1,82 @@
...
+class ServiceError(CoffeeHouseError):
+    """
+    This error can be a generic error,
+    see the error message for more details
+    """
     pass
 
-
-class SessionNotFoundError(CoffeeHouseError):
+class EmptyOutputRequest(Exception):
```
Comment (penn5) [inline coffeehouse/exception.py:76]: Bad naming

File: setup.py [hunk 6]
```diff
@@ -1,22 +1,27 @@
...
+        'coffeehouse': 'coffeehouse',
     },
-    author='Intellivoid Technologies',
+    author=__author__,
     author_email='netkas@intellivoid.net',
     url='https://coffeehouse.intellivoid.net/',
     install_requires=[
-        'requests>=2.3.0'
-    ]
+        'requests>=2.3.0',
```
Comment (penn5) [inline setup.py:25]: Should really use ~=

**Comment by Netkas** [comment]
I'll run a few tests later today and find out if it works as expected.

File: coffeehouse/lydia/session.py [hunk 7]
```diff
@@ -8,10 +8,10 @@ def __init__(self, data, client):
...
 
         self._client = client
-        self.id = data["session_id"]
-        self.language = data["language"]
-        self.available = data["available"]
-        self.expires = data["expires"]
+        self.id = data['session_id']
+        self.language = data['language']
+        self.available = data['available']
+        self.expires = data['expires']
```
Comment (Netkas) [inline coffeehouse/lydia/session.py]: Note that a session now contains more information about the session itself.

https://docs.intellivoid.net/coffeehouse/v1/lydia/get_session
https://docs.intellivoid.net/coffeehouse/v1/lydia/get_session_attributes

Comment (CustomIcon) [reply coffeehouse/lydia/session.py]: OMG>>>>>>

Comment (CustomIcon) [reply coffeehouse/lydia/session.py]: yeah let me add more commits to lydia. i didnt play around it (just my ide auto refractoring this)

**Pull request merged**